lib/actions/build.coffee

```diff
@@ -0,0 +1,81 @@
+# Imported here because it's needed for the setup
+# of this action
+dockerUtils = require('../utils/docker')
+
+exports.getBundleInfo = getBundleInfo = (options) ->
+	Promise = require('bluebird')
+	helpers = require('../utils/helpers')
+
+	if options.application?
+		# An application was provided
+		Promise.try(-> helpers.getAppInfo(options.application))
+		.then (app) ->
+			return [app.arch, app.device_type]
+	else if options.arch? and options.deviceType?
+		return Promise.resolve([options.arch, options.deviceType])
+	else
+		# No information, cannot do resolution
+		return
+
+module.exports =
+	signature: 'build [context]'
+	description: 'Build a container locally'
+	permission: 'user'
+	help: '''
+		Use this command to build a container with a provided docker daemon.
+
+		You must provide either an application or a device-type/architecture
+		pair to use the resin Dockerfile pre-processor
+		(e.g. Dockerfile.template -> Dockerfile).
+
+		Examples:
+
+			$ resin build
+			$ resin build ./context/
+			$ resin build -d raspberrypi3 -A armhf
+			$ resin build -a MyApp ./context/
+			$ resin build --docker '/var/run/docker.sock'
+			$ resin build -h my.docker.host -p 2376 --ca ca.pem --key key.pem --cert cert.pem
+	'''
+	options: dockerUtils.appendOptions [
+		{
+			signature: 'arch'
+			parameter: 'arch'
+			description: 'The architecture to build for'
+			alias: 'A'
+		},
+		{
+			signature: 'deviceType'
+			parameter: 'deviceType'
+			description: 'The type of device this build is for'
+			alias: 'd'
+		},
+		{
+			signature: 'context'
+			parameter: 'context'
+			description: "The build context, which contains the build files. Defaults to '.'"
+			alias: 'c'
+		},
+		{
+			signature: 'application'
+			parameter: 'application'
+			description: 'The target resin.io application this build is for'
+			alias: 'a'
+		},
+		{
+			signature: 'tag'
+			parameter: 'tag'
+			description: 'The alias to the generated image'
+			alias: 't'
+		},
+		{
+			signature: 'noCache'
```

Comment:
is it case-sensitive? Not *nix idiomatic I think, usually all params are lowercase. I don't know if Capitano handles dash-separated names, but `--nocache` also looks fine